lib/sessions.js

```diff
@@ -162,17 +188,17 @@ class ServerSessionPool {
    */
   release(session) {
     const sessionTimeoutMinutes = this.topology.logicalSessionTimeoutMinutes;
-    while (this.sessions.length) {
-      const session = this.sessions[this.sessions.length - 1];
+    while (this.pool.length) {
+      const session = this.pool[this.pool.length - 1];
       if (session.hasTimedOut(sessionTimeoutMinutes)) {
-        this.sessions.pop();
+        this.pool.pop();
       } else {
         break;
       }
     }
 
```

Comment:
I think you're forgetting to remove the session from `active` in this case? Perhaps you're depending on the emitted `ended` event?

Reply:
yeah, im relying on the `ended` event